clickhouse/base/socket.cpp

```diff
@@ -251,6 +251,8 @@ SOCKET SocketConnect(const NetworkAddress& addr) {
                         SetNonBlock(s, false);
                         return s;
                     }
+                    throw std::system_error(
```

Comment:
The code inside "if (connect(...)) ... " block in case of EINPROGRESS waits for 5 sec. and after that tries to realize was the connection established or not. So if we throw an error here we will not try other addresses. Or maybe I missing something?

Reply:
You are right. Thanks for pointing that out.  Will update it to be throw last connection err instead of breaking out the for loop